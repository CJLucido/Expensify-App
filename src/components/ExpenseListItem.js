import React from 'react';
import { Link } from "react-router-dom";
import moment from 'moment';
import numeral from 'numeral';

//for the button in ExpenseListItem.js props.dispatch messes it all up but in ExpenseListFilter.js I can add or remove "props." from before dispatch with no conscequence, why? It is destructured off of props here so you wouldnt call props of props, I dont know why it will work either way in the other, maybe because there is only ever one dispatch?

 const ExpenseListItem = ({ dispatch, id, description, amount, createdAt}) => (  
    <div>
        <Link to={`/poopypooppoop/${id}`}>
        <h3>
        {description}
        </h3>
        </Link>
        <p>
        {numeral(amount / 100).format('$0,0.00')}
         - 
        {moment(createdAt).format('MMMM Do, YYYY')}
        </p>
    </div>
); 


export default ExpenseListItem;

