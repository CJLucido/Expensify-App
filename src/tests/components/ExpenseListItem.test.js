import React from 'react';
import {shallow} from 'enzyme';
import ExpenseListItem from "../../components/ExpenseListItem";
import expenses from "../fixtures/expenses";

test("should render expense list with expense[0]/id 1", () =>{
    const wrapper = shallow(<ExpenseListItem {...expenses[0]}/>);
    expect(wrapper).toMatchSnapshot();
});