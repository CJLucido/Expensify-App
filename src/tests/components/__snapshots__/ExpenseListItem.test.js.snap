// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense list with expense[0]/id 1 1`] = `
<div>
  <Link
    to="/poopypooppoop/1"
  >
    <h3>
      gum
    </h3>
  </Link>
  <p>
    195
     - 
    0
  </p>
</div>
`;
